### ISSUE TITLE
Black screen on TTY1 after booting only

### ISSUE THREAD

**Issue by JGG-Apps** [substring/os#23]
I recently installed from scratch GroovyArcade 2022.10 OS (Latest available) using a 15khz CRT monitor.

The installation was completed very fine, no issue.

After the installation I can see BIOS logo with bad H SYNC ( normal)... Then I see the GroovyArcade booting image with progress bar, then I get a black screen. I heard sound but no screen.

If i try to switch on TTY2,3,4,5,6 with ALT+F2,3,4,5,6... I see correctly the command console. I come back on TTY1 and the black screen still.

I return on TTY2 and I try to kill startfe.sh process... then sometime rhis force to reload the MAME and Isee properly the MAME on TTY1 now. Everything still working fine until I power off or reboot the computer... 

Does a know problem ? New problem with 2022.10 ?

**Comment by JGG-Apps** [comment]
To ensure all my test and configuration was done properly. I decided to download GroovyArcade 2022.01 (previous version) and install it on the same hardware.

I confirm everything is working fine, no issue with 2022.01.

**Comment by substring** [comment]
Hi,

Can you share details about your configuration ? What GPU, if you have an onboard GPU, etc...
The Leon change between the 2 OS releases is that 2022.21 uses KMS by default, and SFML is not working well yet with multiple GPUs

**Comment by JGG-Apps** [comment]
Hi,

My hardware configuration:

Lenovo M72e (PN: 4004) (Intel i3 CPU, Intel HD graphics, Yes onboard GPU)

On this hardware (tested more 1 PC that have the same hardware):
2022.01 with 15 khz VGA-1 = Work fine !
2022.10 with 15 khz = Not work properly (toggle with TTY2 kill process can fix the issue)
2022.10 with 15 khz Intel/nVidia = Not work properly (toggle with TTY2 kill process can fix the issue)
2022.10 with LCD = Not work properly (toggle with TTY2 kill process can fix the issue)

NOTE: Testing this hardware with Debian 9/10/11/12, Ubuntu 18/20, Kali, ParrotOS, Windows, etc. is always working fine

**Comment by substring** [comment]
My guess is that you need X11 instead of KMS. I can't say I encourage Intel or Nvidia GPUs, they give erratic results. Try switching to X11 in gasetup > setup > video setup > X/KMS then just reboot

**Comment by JGG-Apps** [comment]
Hi, thank you for your very fast answer !!! Yes' i can understand to don't encourage Intel/nVidia, but the actual reality on the market i: If you want to use a commercial, robust hardware and in this case tiny format like Lenovo, HP, Intel NUC to easly insert them in a machine (embedded concept)... where these tiny machine don't have expansion slots, you are juste facing using Intel GPU ;)

During this, I decided to test on an older machine I had: Lenovo M58 tower with nVidia card in PCIe slot.

2022.10 is working fine in 15khz mode !!! 

------------------------------

Then I think your idea about X11 vs KMS should be something I need to test. I will confirm you ASAP.

Finally, I have a suggestion to improve GroovyArcade adoption. Actually, it don't have an official website, just the project on GitHub. It may have at least a link to an official web page where you could list "compatible hardware" certified and tested.

**Comment by substring** [comment]
You mean https://gitlab.com/groovyarcade/support/-/wikis/2-Pre-Requisites-and-Installation/2.1-Hardware-Suggestions-General ?

Most of everything happens on Discord, you'll find a link on the wiki

But what I should do is at least add README to that project, yeah

**Comment by JGG-Apps** [comment]
Oh tanks, Your GitLab WIKI is interesting !!!

For this issue. Weird symptoms...

First, I switched to X11. Since I selected X11, on reboot I just get the GroovyArcade logo at boot with progress bar 100% and still there (not freezed, I can see the logo fade in/out)... If I wait 5-10 minutes the logo still there. 

Where it start to be weird:
If I press ESC to see the booting steps, the result is never the same... It could be every step like:
Starting Samba SMB Daemon...
or
[OK] Finished Initializes Pacman keyring

but... If I press again ESC, I get the gasetup telling "Exit Setup Menu ?"

It's like what I get from the first ESC is just a kind of cache.

If I answer "No" to exit setup, return to main menu, select "Start Frond End", MAME Attreact Plus is shown !

-------------------

Aftert that, if I decide to make a clean POWER OFF from gasetup, relaunch the PC, I still getting the same issue.

- I need to wait 2 min least
- I press ESC, the kernel boot process is shown
- I press ESC once again, the gasetup tell me to exit, I say no and I return to main menu to start the Front-End. 
- I can use MAME !

---------------------

Knowing that, I decided to came back with KMS to test same procedure... No luck. I got the initial issue where I have a full black screen with KMS. ESC key not work.

-----------------------

Then,

Yes X11 respond in another manner... I can manually tweak but something still hanging somewhere and not giving an expected result.

In bith case, KMS or X11 something happen where video is not properly displayed "on normal boot process only". If I force (break, kill, relaunch, etc.) the KMS or X11 is displaying the MAME console ?!?

**Comment by JGG-Apps** [comment]
I generated a log: https://transfer.sh/09Y6pm/ga-support.tar.zst

**Comment by substring** [comment]
Make sure you've updated GA, using X11 required some trickery that was fixed after the release of the ISO

**Comment by JGG-Apps** [comment]
Yep, pacman -Syu.... and some problems due tu an expired certificate on 2022-12-01 needs a bypass in pacman.conf. But after that, it's not better.

Obtenir Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by substring** [comment]
you don't bypass anything, you update first archlinux-keyring, then the full OS.

what happens when you start `/opt/galauncher/startfe.sh` ?

**Comment by JGG-Apps** [comment]
No... the certificate I talk was not renewd by the author. We need to ignore the signature for the moment :( archlinux-keyring update not fix... another forums on the Internet talk about this issue for ArchLinux.

Since the initial request, If I kill startfe.sh manually from TTY2, then try to launch startfe.sh manually, it always work fine.



Obtenir Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by substring** [comment]
What happens when you mannually launch startfe.sh when X is set as the video backend ? do you get any error ?

By the way, it's necessary to rollback mesa, 22.3.x broke everything. Just `sudo pacman -U --noconfirm https://archive.archlinux.org/packages/m/mesa/mesa-22.2.4-1-x86_64.pkg.tar.zst`

**Comment by JGG-Apps** [comment]
Hi,

Even if I try to use X11 or KMS, after boot I always get a black screen but I heard soound like the Attract-Mode is running.

I verified and mesa package is OK... I resintalled many many time. Format HDD, reinstall on many different computers, I have over 20 units of this model ;)

But, I found something that could put you on the touch:

After booting, when I heard sound and a black screen, I go trough any other console (TTY2 or higher). I can see the login prompt perfectly (in 15 khz, LCD or any mode depending of my test scenario). When the login prompt is visible, I don't need to open a session. If I just press the ENTER key... this will do something (I don't know why) but I see the mouse cursor for few milliseconds and the MAME console appears on the screen and I'm OK !!!

 Then, I just realize that everything I did on any TTY2 or more console, the ENTER button is what is fixing the issue. It's weird, but it's like the ENTER button still linked with all consoles to force something to keep alive the MAME console ?!

Remember, it's something happen with version 2022.10 only and Intel VGA Onboard video card.

To find the problem, I would like to determine at whiche level exactly this happen. Could you help me to make a tweak ? Could you tell me what I need to set in GroovyArcade OS 2022.10 to don't load anything at boot time... Just boot on a Linux console ? 

The goal is to determine if the issue is caused by KMS or by something loaded after.

**Comment by substring** [comment]
I'd need the output of a few commands:
```bash
for p in /sys/class/drm/card? ; do
  id=$(basename `readlink -f $p/device`)
  name=$(lspci -mms $id | cut -d '"' -f4,6 --output-delimiter=" ")
  cardnum=$(basename $p)
  module_name=$(lspci -ks $id | grep "Kernel driver in use:" | cut -d ":" -f 2 |tr -d ' ')
  echo "$cardnum ($module_name): $name"
  for p in /sys/class/drm/${cardnum}-*/status; do
    con=${p%/status}
    bus=$(ls -d "$con"/i2c-* 2>/dev/null)
    echo -n "  ${con#*/card?-}: $(cat $p)"
    [[ -n $bus ]] && echo -n " - Has i2c"
    edid_size=$(cat "$con"/edid | wc -c)
    if [[ $edid_size > 0 ]] ; then
      echo -n " - Found an EDID with monitor model: "
      edid-decode < "$con"/edid |grep "Display Product Name: " | sed "s/Display Product Name\: //"
    fi
    echo -e "\nList of video modes:"
    cat "$con"/modes
  done
done
```

And as root, when AM is running on KMS:
```bash
grep -E '^'$'\t''(allocated|size)' /sys/kernel/debug/dri/0/framebuffer
```

Your symptoms are like someone who has 2 GPUs. Remember that for X you must update your OS or it won't work.

To boot to a console, just disable the startefe.sh script in ~/.bash_profile. GA is meant to autologin on tty1, disable the service eventually

**Comment by substring** [comment]
closing since it was fixed with the author
